lang/java/reef-experimental/src/main/java/org/apache/reef/experimental/ParquetReader.java

```diff
@@ -0,0 +1,176 @@
+import static org.apache.parquet.format.converter.ParquetMetadataConverter.NO_FILTER;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+
+/**
+ * A reader for Parquet files that can serialize data to local disk and return Avro schema and Avro reader.
+ * The intent is not to build a general parquet reader, but to consume data with table-like property.
+ */
+public final class ParquetReader {
+  /**
+   * Standard java logger.
+   */
+  private static final Logger LOG = Logger.getLogger(ParquetReader.class.getName());
+  
+  private Path parquetFilePath;
+
+  @NamedParameter(default_value="", doc="Path to parquet file", short_name="path")
+  class PathString implements Name<String> { }
+
+  @Inject
+  private ParquetReader(@Parameter(ParquetReader.PathString.class) final String path) throws IOException {
+    parquetFilePath = new Path(new File(path).getAbsolutePath());
+    final Schema schema = getAvroSchema();
+    if (schema.getType() != Schema.Type.RECORD) {
+      LOG.log(Level.SEVERE, "ParquetReader only support Avro record type that can be consumed as a table.");
+    }
+    for (final Schema.Field f : schema.getFields()) {
+      if (f.schema().getType() == Schema.Type.RECORD) {
+        LOG.log(Level.SEVERE, "ParquetReader doesn't support nested record type for its elements.");
+      }
+    }
+  }
+
+  /**
+   * Retrieve avro schema from parquet file.
+   * @return avro schema from parquet file.
+   * @throws IOException if the Avro schema couldn't be parsed from the parquet file.
+   */
+  public Schema getAvroSchema() throws IOException {
+    return getAvroSchema(new Configuration(true), NO_FILTER);
+  }
+  
+  /**
+   * Retrieve avro schema from parquet file.
+   * @param configuration Hadoop configuration.
+   * @param filter Filter for Avro metadata.
+   * @return avro schema from parquet file.
+   * @throws IOException if the Avro schema couldn't be parsed from the parquet file.
+   */
+  public Schema getAvroSchema(final Configuration configuration, final MetadataFilter filter) throws IOException {
+    final ParquetMetadata footer = ParquetFileReader.readFooter(configuration, parquetFilePath, filter);
+    final AvroSchemaConverter converter = new AvroSchemaConverter();
+    final MessageType schema = footer.getFileMetaData().getSchema();
+    return converter.convert(schema);
+  }
+  
+  /**
+   * Construct an avro reader from parquet file.
+   * @return avro reader based on the provided parquet file.
+   * @throws IOException if the parquet file couldn't be parsed correctly.
+   */
+  public AvroParquetReader<GenericRecord> getAvroReader() throws IOException {
```

Comment:
please rename to `createAvroReader()` as this is not a getter but creates a new object. Same in other instances, if I miss them.